src/core/server/saved_objects/migrations/model/model.ts

```diff
@@ -72,7 +72,26 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   if (stateP.controlState === 'INIT') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
 
-    if (Either.isRight(res)) {
+    if (Either.isLeft(res)) {
+      const left = res.left;
+      if (isLeftTypeof(left, 'unsupported_cluster_routing_allocation')) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          reason: `The elasticsearch cluster has cluster routing allocation incorrectly set for migrations to continue. To proceed, please remove the cluster routing allocation settings with PUT /_cluster/settings {"transient": {"cluster.routing.allocation.enable": null}, "persistent": {"cluster.routing.allocation.enable": null}}`,
```

Comment:
Updated the reason and message to more accurately describe that the settings aren't supported in migrations.